notebooks/problem-sets/quantum_enigma_problem_set.ipynb

```diff
@@ -0,0 +1,299 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Quantum Enigma 001 - The Treasure Door Problem Set"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "![University of Sherbrooke, Institute of Quantique](images/IQ_Logo.png)\n",
+    "\n",
+    "## Overview\n",
+    "\n",
+    "Watch the following video before attempting this problem set\n",
+    "\n",
+    "<div class=\"youtube-wrapper\">\n",
+    "    <iframe width=\"560\" height=\"315\" src=\"https://www.youtube.com/embed/c1beJIg8lRs\" title=\"YouTube video player\" frameborder=\"0\" allow=\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture\" allowfullscreen></iframe>\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<!-- ::: q-block.exercise -->\n",
+    "\n",
+    "### Quick quiz\n",
+    "\n",
+    "<!-- ::: q-quiz(goal=\"enigma-quiz-0\") -->\n",
+    "\n",
+    "<!-- ::: .question -->\n",
+    "\n",
+    "Ready for your quiz?\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option(correct) -->\n",
+    "\n",
+    "1. Yes -- Let's go! \n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option -->\n",
+    "\n",
+    "2. Not yet, I need to watch the videos again.\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: -->"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Problem 1\n",
+    "\n",
+    "Sometimes a quantum circuit can be simplified. One way of achieving this is by cancelling some quantum gates. Could you simplify the following circuit?"
```

Comment:
Would it be possible to add an option to draw the circuit to help visualise what it looks like mid-way through the process? in another box maybe?

Reply:
Yep definitely. Will revisit tonight!

Reply:
Added as a `qc.draw()` in the subsequent cell